metals/src/main/scala/scala/meta/metals/providers/DocumentFormattingProvider.scala

```diff
@@ -63,14 +63,16 @@ class DocumentFormattingProvider(
       }
       .toFunction0()
 
-  private val fullDocumentRange = Range(
-    start = Position(0, 0),
-    end = Position(Int.MaxValue, Int.MaxValue)
-  )
-
   def format(
       input: Input.VirtualFile
   ): Task[Either[Response.Error, List[TextEdit]]] = Task {
+    val fullDocumentRange = Range(
+      start = Position(0, 0),
+      end = Position(
+        input.chars.count(_ == '\n'),
```

Comment:
This will fail for `\r\n` line breaks, we can use `scala.meta.Position.Range` instead

```scala
    val pos = scala.meta.Position.Range(input, 0, input.chars.length)
    (pos.startLine, pos.startColumn, pos.endLine, pos.endColumn)
```